simulation/behavior_tree_plugin/src/vehicle/follow_lane_sequence/stop_at_traffic_light_action.cpp

```diff
@@ -102,13 +105,12 @@ BT::NodeStatus StopAtTrafficLightAction::tick()
   if (waypoints.waypoints.empty()) {
     return BT::NodeStatus::FAILURE;
   }
-  const auto spline = traffic_simulator::math::CatmullRomSpline(waypoints.waypoints);
   const auto distance_to_traffic_stop_line =
-    hdmap_utils->getDistanceToTrafficLightStopLine(route_lanelets, waypoints.waypoints);
+    hdmap_utils->getDistanceToTrafficLightStopLine(route_lanelets, *trajectory);
   if (!distance_to_traffic_stop_line) {
     return BT::NodeStatus::FAILURE;
   }
-  distance_to_stop_target_ = getDistanceToTrafficLightStopLine(route_lanelets, waypoints.waypoints);
+  distance_to_stop_target_ = getDistanceToTrafficLightStopLine(route_lanelets, *trajectory);
```

Comment:
Currently, *trajectory cannot be null pointer.
However, please do null check for future development.